File: java/snakes-and-ladders/src/main/java/gagan93jtg/lld/models/Ladder.java
package gagan93jtg.lld.models;

public class Ladder {
  private int start;
  private int end;

  public Ladder(int start, int end) {
    this.start = start;
    this.end = end;

    validateLadder();
  }

  public int getStart() {
    return start;
  }

  public int getEnd() {
    return end;
  }

  private void validateLadder() {
    if (start > end) {
      throw new IllegalArgumentException("A ladder should start at a smaller value and end at higher value");
    }
  }
}

File: ruby/snakesladders/lib/snakesladders/models/ladder.rb
module Snakesladders
  module Models
    class Ladder
      attr_accessor :start_point, :end_point

      def initialize(start_point, end_point)
        self.start_point = start_point
        self.end_point = end_point

        validate_ladder
      end

      private def validate_ladder
        raise ::Snakesladders::Errors::InvalidLadderError.new("A ladder should start at a smaller value and end at higher value") if start_point > end_point
      end
    end
  end
end
